enhancements/builds/remove-jenkins-playload.md

```diff
@@ -0,0 +1,225 @@
+
+Jenkins itself is not native to Kubernetes and does not use a Java runtime that has been optimized for cloud-native environments (i.e. Quarkus).
+As a result, it has poor resilience when run on Kubernetes.
+The Jenkins templates do not run Jenkins in a highly available mode, leaving its primary instance vulnerable to downtime from slow restarts.
+The Jenkins container image is also very large - it is in fact the largest image in the OpenShift payload today.
+Removing Jenkins from the OpenShift payload has been a longstanding goal that can reduce the time needed to pull the OCP payload at install/updgrade time.
+
+With the GA release of OpenShift Pipelines, we would like to decouple Jenkins from OpenShift, distributing Jenkins just like any other application.
+This would allow bug fixes in our Jenkins distribution to be released on its own cadence.
+Customers can continue to deploy Jenkins on OpenShift via its ImageStream and associated templates.
+In this fashion, Jenkins becomes just another application on OpenShift and does not retain any preferential status.
+The maintenance of Jenkins is also simplified - only one image version will be released and tested against supported OCP versions.
+
+### Goals
+
+* Move Jenkins out of the OCP Payload.
+
+### Non-Goals
+
+* Document how JenkinsPipeline strategy builds can be migrated off of OpenShift.
+* Migrate Jenkins users to OpenShift Pipelines/Tekton.
+* Stop distributing the Jenkins image and our associated OpenShift plugins.
+* Disable the JenkinsPipeline build strategy.
+
+## Proposal
+
+### User Stories
+
+As a developer using Jenkins on OpenShift
+I want to continue using my Jenkinsfiles
+So that I can continue to use Jenkins for my CI/CD processes
+
+As an OpenShift release engineer
+I want to remove Jenkins from the OCP payload
+So that I can reduce the size of the payload
+And release Jenkins fixes on their own cadence
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Remove From Payload
+
+Jenkins and its related agent images will be removed from the OCP payload.
+The delivery of these container images will be replaced with a CPaaS-based release pipeline.
+Jenkins images will be published to the Red Hat Container Catalog, just like any other layered product.
+
+Once the CPaaS pipeline is ready, the Jenkins templates and imagestreams delivered in OpenShift via Samples operator will need to reference the container catalog images (just like other layered products).
+The Samples Operator can then remove its special logic for Jenkins imagestreams and templates.
+
+#### Jenkins Image Tagging Scheme
+
+The Jenkins image will use the maximum supported OCP version as the basis of its tagging scheme:
+
+1. `latest` - latest image build
+2. `4` - latest image that supports OCP 4
+3. `4.<Y>`- latest image that supports up to OCP 4.Y
+4. `4.<Y>.<Z>` - latest image that supports up to OCP 4.Y, with bugfix advisory Z
+
```

Comment:
@akram I'm a bit naive when it comes to what CPaaS can do. Can we configure it to produce multiple tags from the same branch?

For example - if the image in main/master supports OCP 4.6, can we configure CPaaS to produce the following tags:

1. `latest`, `4`
2. `4.8`, `4.8.0`
3. `4.7`, `4.7.<z>`
4. `4.6`, `4.6.<z>`

Similarly, when we branch for backwards compatibility, can we tell CPaaS to produce a different set of tags on the backport branch?

Reply:
Reviewed with the CPaaS folks - the answer is yes, sort of. CPaaS + Errata lets us pick which tags get pushed to the container catalog.